crates/rooch-types/src/address.rs

```diff
@@ -303,23 +311,48 @@ impl TryFrom<MultiChainAddress> for EthereumAddress {
 }
 
 #[derive(Debug, Clone, PartialEq, Eq, Hash)]
-pub struct BitcoinAddress(pub bitcoin::Address);
+pub struct BitcoinAddress(pub Address);
 
 impl RoochSupportedAddress for BitcoinAddress {
-    fn random() -> Self {
-        Self(bitcoin::Address::new(
-            bitcoin::Network::Bitcoin,
-            bitcoin::address::Payload::PubkeyHash(PubkeyHash::from_raw_hash(
-                hash160::Hash::from_slice(H160::random().as_bytes()).unwrap(),
-            )),
-        ))
+    fn random(protocols: &Protocols) -> Self {
```

Comment:
Do not put protocols as the argument, random the protocol in this function.

Reply:
Updated.